L1Trigger/Phase2L1Taus/interface/L1HPSPFTauBuilder.h

```diff
@@ -0,0 +1,118 @@
+#ifndef L1Trigger_Phase2L1Taus_L1HPSPFTauBuilder_h
+#define L1Trigger_Phase2L1Taus_L1HPSPFTauBuilder_h
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"                   // edm::ParameterSet
+#include "DataFormats/Provenance/interface/ProductID.h"                   // edm::ProductID
+
+#include "L1Trigger/Phase2L1Taus/interface/LocalFileInPath.h"          // LocalFileInPath
+#include "L1Trigger/Phase2L1Taus/interface/L1HPSPFTauQualityCut.h" // L1HPSPFTauQualityCut
+#include "DataFormats/Phase2L1Taus/interface/L1HPSPFTau.h"         // l1t::L1HPSPFTau
+#include "DataFormats/L1TParticleFlow/interface/PFCandidate.h"       // l1t::PFCandidate, l1t::PFCandidateCollection, l1t::PFCandidateRef
+#include "DataFormats/L1TParticleFlow/interface/PFJet.h"             // l1t::PFJet, l1t::PFJetCollection, l1t::PFJetRef
+#include "DataFormats/L1TCorrelator/interface/TkPrimaryVertex.h"
+
+#include <TFormula.h> // TFormula
+#include <TFile.h>    // TFile
+#include <TH1.h>      // TH1
+
+#include <vector>
+
+class L1HPSPFTauBuilder
+{
+ public:
+  L1HPSPFTauBuilder(const edm::ParameterSet& cfg);
+  ~L1HPSPFTauBuilder();
+  
+  void reset();
+  void setL1PFCandProductID(const edm::ProductID& l1PFCandProductID);
+  //void setVertex(const l1t::VertexRef& primaryVertex);
+  void setVertex(const l1t::TkPrimaryVertexRef& primaryVertex);
+  void setL1PFTauSeed(const l1t::PFCandidateRef& l1PFCand_seed);
+  void setL1PFTauSeed(const l1t::PFJetRef& l1PFJet_seed);
+  void addL1PFCandidates(const std::vector<l1t::PFCandidateRef>& l1PFCands);
+  void setRho(double rho);
+  void buildL1PFTau();
+
+  l1t::L1HPSPFTau getL1PFTau() const { return l1PFTau_; }
+
+ private:
+  l1t::PFCandidateRefVector convertToRefVector(const std::vector<l1t::PFCandidateRef>& l1PFCands);
+
+  bool isWithinSignalCone(const l1t::PFCandidate& l1PFCand);
+  bool isWithinStrip(const l1t::PFCandidate& l1PFCand);
+  bool isWithinIsolationCone(const l1t::PFCandidate& l1PFCand);
+
+  TFormula* signalConeSizeFormula_;
+  static int signalConeSizeFormula_instance_counter_;
+  double signalConeSize_;
+  double signalConeSize2_;
+  double min_signalConeSize_;
+  double max_signalConeSize_;
+
+  bool useStrips_;
+  double stripSize_eta_;
```

Comment:
general comment, please use camelCase rather_than_underscores and pleaseDont_mix_them. 

https://cms-sw.github.io/cms_coding_rules.html